plugin/helpers_kv.go

```diff
@@ -220,3 +220,25 @@ func (p *HelpersImpl) KVListWithOptions(options ...KVListOption) ([]string, erro
 
 	return ret, nil
 }
+
+func (p *HelpersImpl) KVModify(key string, f func(value []byte) ([]byte, error)) error {
+	err := p.ensureServerVersion("5.2.0")
+	if err != nil {
+		return err
+	}
+
+	v, appErr := p.API.KVGet(key)
+	if appErr != nil {
+		return appErr
+	}
+
+	modifiedValue, modifiedError := f(v)
+	if modifiedError != nil {
+		return err
+	}
+	kvErr := p.API.KVSet(key, modifiedValue)
```

Comment:
```suggestion
	appErr = p.API.KVSet(key, modifiedValue)
```

Reply:
Spoke to @hanzei offline and instead of `KVSet` we want to use `KVSetWithOptions` to make this operation atomic.

Reply:
Thank you for the comments @ali-farooq0. Just finish resolving all of them.